### ISSUE TITLE
Misinterpreted RawSocket message length header bytes

### ISSUE THREAD

**Issue by ecorm** [crossbario/crossbar#1499]
My colleague corresponded with you previously about a possible problem with Crossbar rawsockets for large messages.

There indeed was a problem on our end in CppWAMP, but the Crossbar problem still persists after resolving it. I have manually inspected the 32-bit RawSocket header and message in Wireshark, and Crossbar gets confused about the expected message length.

Header as it appears in Wireshark: `00 02 08 2c` -> 133164 decimal
Length of message in Wireshark's Follow TCP Stream: Exactly 133164 bytes

Crossbar error message: `failing RawSocket connection - message length exceeded: message was 133164 bytes, but current maximum is 131072 bytes`

131072 is exactly 128 x 1024. Crossbar seems to be zeroing the two least significant bytes when it computes its "current maximum" of 131072.

If it matters, Wireshark shows that the long message was split across a number of TCP segments.

This is a showstopper for us: it prevents us from scaling to larger buildings in our building automation system.

I'll soon release a CppWAMP version where anyone can duplicate the problem by running CppWAMP's unit test.

**Comment by ecorm** [comment]
Please standby, this may be a false alarm. I think I mis-configured Crossbar's maximum message size during my testing.

**Comment by ecorm** [comment]
False alarm: I did indeed mis-configuring Crossbar's maximum message size.

The good news is that I confirmed my colleague's findings that the problem was on our end with CppWAMP. I was able to create a test case directly in CppWAMP that duplicates the jumbled RawSocket message (instead of relying on our app to cause the failure). I have fixed the problem in CppWAMP and will publish it soon.

TLDR: This had nothing to do with Crossbar. Sorry for the noise.
